### ISSUE TITLE
SSL certificate problem when trying to get data from remote collection

### ISSUE THREAD

**Issue by Lloydinator** [tightenco/jigsaw#486]
So I'm trying to get a collection from Contentful but I'm getting an error that says `cURL error 60: SSL certificate problem: unable to get local issuer certificate (see https://curl.haxx.se/libcurl/c/libcurl-errors.html)`.

Research indicates that the problem is related to Guzzle but I want to know how I could solve this problem without having to go into my vendor folder and manually edit the files.

**Comment by bakerkretzmar** [comment]
What version of Jigsaw and guzzle are you on? Are you on Windows or is this happening on a server?

This is most likely an issue with the machine you're running Jigsaw on, I don't think it's curl or guzzle and it's definitely not Jigsaw. You'll probably need to update your system or take a closer look in your `php.ini`.

See:
- https://laracasts.com/discuss/channels/general-discussion/curl-error-60-ssl-certificate-problem-unable-to-get-local-issuer-certificate?page=2
- guzzle/guzzle#1855
- https://thisinterestsme.com/php-curl-ssl-certificate-error
- https://stackoverflow.com/q/24611640
- https://serverfault.com/q/755248
- https://stackoverflow.com/q/28858351

**Comment by Lloydinator** [comment]
I do think it could be my machine, but I tried it on a newer machine and the same thing. Also, I'm not using xampp or wamp to serve the app so I don't see how editing php.ini would work (I did try btw).

**Comment by Lloydinator** [comment]
Oh and my Jigsaw version is 1.3 and Guzzle is 6.5.5 @bakerkretzmar

**Comment by bakerkretzmar** [comment]
@Lloydinator are you seeing this error locally or on a server, and what OS is it on? If you're on macOS, are you using Valet?

**Comment by Lloydinator** [comment]
I'm not using any of that because as far as I see it the build command for Jigsaw doesn't use Valet or Homestead or any of that. I'm seeing the error locally. I'm thinking of just editing the Guzzle library but they say editing vendor manually is a bad idea and plus it wouldn't carry over into production.

**Comment by Lloydinator** [comment]
Ok so I did further research into the error and it says the error will happen if "The remote server's SSL certificate or SSH md5 fingerprint was deemed not OK." I think my connection with Contentful could be the cause? Hmm I think I need to review my connection to the API. I thought the Contentful SDK would've thrown its own errors.

**Comment by bakerkretzmar** [comment]
@Lloydinator what operating system are you using?

**Comment by Lloydinator** [comment]
W10. My thinking is now to use this source: https://visdap.blogspot.com/2019/01/curl-error-60-ssl-certificate-in.html, and use solution 5 to write some code for `beforeBuild`, or maybe even just putting it into the config file. I'll try myself but any help with doing that would be appreciated.

**Comment by Lloydinator** [comment]
Ok I reinstalled wampserver and PHP and edited the php.ini file to use the 'cacert.pem' file as suggested by the links you posted and it works now. I did try editing php.ini earlier and it didn't work. I'm thinking that before I might have not put the certificate in the correct php version, since there are more than one on my machine. Could also be some other stuff that the wampserver reinstall solved. Anyway, I'm just glad I can move on.
